iohub/ngff/utils.py

```diff
@@ -0,0 +1,485 @@
+    """
+
+    # Limiting the chunking to 500MB
+    if chunks is None:
+        chunk_zyx_shape = _calculate_zyx_chunk_size(
+            shape,
+            np.dtype(dtype).itemsize,
+            max_chunk_size_bytes,
+        )
+
+        chunks = 2 * (1,) + chunk_zyx_shape
+
+    # Create plate
+    output_plate = open_ome_zarr(
+        str(store_path), layout="hcs", mode="a", channel_names=channel_names
+    )
+
+    # Create positions
+    for position_key in position_keys:
+        position_key_string = "/".join(position_key)
+        # Check if position is already in the store, if not create it
+        if position_key_string not in output_plate.zgroup:
+            position = output_plate.create_position(*position_key)
+            _ = position.create_zeros(
+                name="0",
+                shape=shape,
+                chunks=chunks,
+                dtype=dtype,
+                transform=[TransformationMeta(type="scale", scale=scale)],
+            )
+        else:
+            position = output_plate[position_key_string]
+
+    # Check if channel_names are already in the store, if not append them
+    for channel_name in channel_names:
+        metadata_channel_names = position.channel_names
+        if channel_name not in metadata_channel_names:
+            position.append_channel(channel_name, resize_arrays=True)
+
+
+def apply_transform_to_zyx_and_save(
+    func: Callable,
+    position_key: str,
+    input_store_path: Path,
+    output_store_path: Path,
+    input_channel_indices: Union[list[slice], list[list[int]]],
+    output_channel_indices: Union[list[slice], list[list[int]]],
+    input_time_indices: int,
+    output_time_indices: int,
+    **kwargs,
+) -> None:
+    """
+    Load a CZYX array from a Position object,
+    apply a transformation, and save the result.
+
+    Parameters
+    ----------
+    func : Callable
+        The function to be applied to the data.
+        Should take a CZYX array and return a transformed CZYX array.
+    position_key : str
+        The label of the position to process, e.g. "A/1/0".
+    input_store_path : Path
+        The path to input OME-Zarr Store.
+    output_store_path : Path
+        The path to output OME-Zarr Store.
+    input_channel_indices : Union[list[slice], list[list[int]]]
+        The channel indices to process. Acceptable values:
+        - A list of slices: [slice(0, 2), slice(2, 4), ...].
+        - A list of lists of integers: [[0], [1], [2, 3, 4], ...].
+        If empty list, process all channels.
+    output_channel_indices : Union[list[slice], list[list[int]]]
+        The channel indices to write to. Acceptable values:
+        - A list of slices: [slice(0, 2), slice(2, 4), ...].
+        - A list of lists of integers: [[0], [1], [2, 3, 4], ...].
+        If empty list, write to all channels.
+    input_time_indices : int
+        The time index to process.
+    output_time_indices : int
+        The time index to write to.
```

Comment:
These should be `input_time_index` and `output_time_index`, correct?

Will these ever be different? If you added both arguments for symmetry with `input/output_channel_indices`, then should we add `input/output_position_key` (or drop `position_key` and instead change `input/output_store_path` to `input/output_position_path`).